src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.Literal.cs

```diff
@@ -194,6 +194,20 @@ public void WriteBoolean(ReadOnlySpan<byte> utf8PropertyName, bool value)
             _tokenType = value ? JsonTokenType.True : JsonTokenType.False;
         }
 
+        internal void WriteBooleanEscapeValueOnly(ReadOnlySpan<byte> escapedUtf8PropertyName, bool value)
```

Comment:
I think we should avoid such tight coupling between the writer and serializer. Generally speaking, the serializer was meant to be written on top of the **public** surface area of the writer layer. If the public overloads that take `JsonEncodedText` do not provide similar perf gains, we need to address that implementation. That said, we can make exceptions, but I am not sure this scenario justifies it.

See: https://github.com/dotnet/corefx/blob/d58a8ade31b4731759c92df00d0f6398c52f3d83/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.Literal.cs#L114-L126

Reply:
https://github.com/dotnet/corefx/pull/37909